### ISSUE TITLE
[PDFViewer] Text search is very slow

### ISSUE THREAD

**Issue by tlueder** [syncfusion/flutter-widgets#115]
Hi,

Thanks for your great work on your Flutter widgets.

I've tried the example app on a virtual device and pixel 5 and searching for the "the" takes a long time and the app is not responding for that time. I also saw errors from the flutter sdk that the main thread is too busy.

Now I wonder, did I do something wrong? Or is this normal?

And as es feature request I'd like to ask if you could add an indexer which runs after the pdf was loaded.
This should help with searching a lot.

**Comment by ramkumarsync** [comment]
Hi @tlueder,

Greetings from Syncfusion.

Please find our comments from below,

I've tried the   example app on a virtual device and pixel 5 and searching for the "the" takes a long time and the app is not responding for that   time. I also saw errors from the flutter sdk that the main thread is too busy.  Now I wonder, did I  do something wrong? Or is this normal? | For now, we have implemented our text search to work synchronously and that causes the delay   and application freeze. We are currently working on modifying our text search   implementation to work asynchronously and this should resolve your problem. This implementation change will be included in our upcoming **2021 Volume 1** release,   which is expected to be available by March 2021.
And as es feature   request I'd like to ask if you could add an indexer which runs after the pdf   was loaded. This should help with searching a lot. | We regret that, we could not understand your requirement. Kindly explain your requirement elaborately   along with the use case scenario to analyze more and assist you with the   better solution at the earliest.

Please let us know if you have concerns.

Regards,
Ramkumar

**Comment by tlueder** [comment]
Hi @ramkumarsync ,

Thanks for your fast response.

Having the search running asynchronously would help a lot with the user experience.

As for my feature request, I had something like indexed search in mind.
https://en.wikipedia.org/wiki/Search_engine_indexing

Indexing all the content asynchronously after the PDF is loaded. 
So when a user searches something and/or changes the search the PDF doesn't need to be parsed again. Only the index.

Let me know what you think about that or if it is out of scope.

Thanks
Thomas

**Comment by dillibabun** [comment]
Hi @tlueder,

Thank you for your response.

We will analyze the feasibility of indexed searching in SfPdfViewer Widget and share the validation details on or before 20 January 2020.

Regards,
Dilli babu.

**Comment by dillibabun** [comment]
Hi @tlueder ,

Indeed, we have extracted the content from PDF document only while loading the document. Then, we are just the finding the text from the PdfTextExtractor instance of Syncfusion Flutter PDF. Once the text searching is made asynchronous, you could see better user experience. Moreover, indexing all content in memory of Flutter App will result in high memory consumption based on document size. We request you to wait till the feature “Asynchronous Text Searching” is included in our upcoming main release 2021 Volume 1.

Regards,
Dilli babu.

**Comment by dillibabun** [comment]
Hi @tlueder ,


We are unable to provide the requested feature "Asynchronous text search". We have used text search methods from Syncfusion PDF package. Since, the requested support is yet to be given in that package, we are unable to do in our widget. As an alternative, we tried to use the Isolate.spawn but, the spawn method allows static members and top-level function to make synchronous methods as asynchronous. We are trying other possible solutions to overcome this issue and will update here once we find the solution for this issue.


Regards,

Dilli babu.
